pepper-apis/src/main/java/org/broadinstitute/ddp/pex/TreeWalkInterpreter.java

```diff
@@ -199,6 +203,19 @@ private boolean applyFormPredicate(InterpreterContext ictx, FormPredicateContext
         }
     }
 
+    private boolean evalStudyQuery(InterpreterContext ictx, StudyQueryContext ctx) {
+        String umbrellaStudyGuid = extractString(ctx.study().STR());
+        return applyStudyPredicate(ictx, ctx.studyPredicate(), umbrellaStudyGuid);
+    }
+
+    private boolean applyStudyPredicate(InterpreterContext ictx, StudyPredicateContext predCtx, String umbrellaStudyGuid) {
+        Optional<GovernancePolicy> policy = ictx.getHandle().attach(StudyGovernanceDao.class).findPolicyByStudyGuid(umbrellaStudyGuid);
+        String msg = "Governance policy for " + umbrellaStudyGuid + " doesn't exist";
+        return policy.map(
+                p -> p.hasReachedAgeOfMajority(ictx.getHandle(), new TreeWalkInterpreter(), ictx.getUserGuid())
```

Comment:
Do you think we can reuse `this` interpreter?

Reply:
To be honest, I considered it but felt uneasy to reuse the same instance due to concerns that such a recursive use can mess up the `TreeWalkInterpreter`'s state.